woocommerce/payment-gateway/assets/js/admin/sv-wc-payment-gateway-token-editor.coffee

```diff
@@ -27,37 +27,47 @@ jQuery( document ).ready ($) ->
 
 		e.preventDefault()
 
+		return unless confirm( wc_payment_gateway_token_editor.actions.remove_token.ays )
+
 		editor = $( this ).closest( 'table' )
 
 		editor.block( message: null, overlayCSS: background: '#fff',opacity: 0.6 )
 
-		if ( confirm( wc_payment_gateway_token_editor.actions.remove_token.ays ) )
+		editor.find( '.error' ).remove()
+
+		row = $( this ).closest( 'tr' )
+
+		# if this is an unsaved token, just remove the row
+		return row.remove() if row.hasClass( 'new-token' )
+
+		data =
+			action:   'wc_payment_gateway_' + editor.data( 'gateway-id' ) + '_admin_remove_payment_token'
+			user_id:  $( this ).data( 'user-id' )
+			token_id: $( this ).data( 'token-id' )
+			security: wc_payment_gateway_token_editor.actions.remove_token.nonce
+
+		$.post wc_payment_gateway_token_editor.ajax_url, data
+
+			.done ( response ) =>
+
+				return handleError( editor, response.data ) unless response.success
 
-			row = $( this ).closest( 'tr' )
+				$( row ).remove()
 
-			# If this is an unsaved token, just remove the row
-			if ( row.hasClass( 'new-token' ) )
+				# no more tokens? Display a message
+				if ( editor.find( 'tr.token' ).length == 0 )
```

Comment:
`is` is preferred over `==` as per https://github.com/polarmobile/coffeescript-style-guide/blob/master/README.md